src/App.tsx

```diff
@@ -1,61 +1,122 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { getNewTodoId, getUserById } from './components/helpers/helpers';
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(usersFromServer, todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(preparedTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [isTitleValid, setIsTitleValid] = useState(false);
+  const [isUserValid, setIsUserValid] = useState(false);
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!userId) {
+      setIsUserValid(true);
+    }
+
+    if (!title.trim()) {
+      setIsTitleValid(true);
+    }
+
+    if (!userId || !title.trim()) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getNewTodoId(todos),
+      completed: false,
+      user: getUserById(usersFromServer, Number(userId)),
+      title: title.trim(),
+      userId: Number(userId),
+    };
+
+    setTodos([...todos, newTodo]);
```

Comment:
when we need our `prevState`, we should pass a callback function to `setState `
```suggestion
    setTodos((prevTodos) => [...prevTodos, newTodo]);
```